lang/src/org/partiql/lang/eval/visitors/StaticTypeInferenceVisitorTransform.kt

```diff
@@ -530,120 +529,74 @@ internal class StaticTypeInferenceVisitorTransform(
-                                                currentType.rangeConstraint == IntType.IntRangeConstraint.UNCONSTRAINED -> currentType
-                                                lastType.rangeConstraint.numBytes > currentType.rangeConstraint.numBytes -> lastType
-                                                else -> currentType
-                                            }
-                                        }
-                                        is FloatType -> StaticType.FLOAT
-                                        is DecimalType -> StaticType.DECIMAL // TODO:  account for decimal precision
-                                        // only missing and null are possible because of argTypeCheck()
-                                        else -> currentType
-                                    }
-                                }
-                                is FloatType -> {
-                                    when (lastType) {
-                                        is IntType -> StaticType.FLOAT
-                                        is FloatType -> StaticType.FLOAT
-                                        is DecimalType -> StaticType.DECIMAL // TODO:  account for decimal precision
-                                        else -> currentType
-                                    }
-                                }
-                                is DecimalType -> {
-                                    when (lastType) {
-                                        is IntType -> StaticType.DECIMAL // TODO:  account for decimal precision
-                                        is FloatType -> StaticType.DECIMAL // TODO:  account for decimal precision
-                                        is DecimalType -> StaticType.DECIMAL // TODO:  account for decimal precision
-                                        else -> currentType
-                                    }
-                                }
-                                else ->
-                                    // this should not be reached, only exists to segment the logic.
-                                    error(
-                                        "Internal Error: SUM function only support Number Type. " +
-                                            "This probably indicates an bug in type inferencer."
-                                    )
-                            }
-                        }
-                    }
-                    // continuation in case of data type mismatch
-                    else {
-                        val expectedType = StaticType.unionOf(StaticType.MISSING, StaticType.NULL, StaticType.NUMERIC)
-                        handleInvalidInputTypeForAggFun(sourceLocation, aggFunc, argType, expectedType)
-                        expectedType
+                // "avg" returns DECIMAL or NULL
+                "avg" -> when {
+                    elementTypes.all { !it.isUnknownOrNumeric() } -> {
```

Comment:
nit: can adjust the boolean expression to be less confusing

```suggestion
                    elementTypes.none { it.isUnknownOrNumeric() } -> {
```